Issue3538 air2air recovery

SenHuang19 opened the pull request:
@JayHuLBL Jianjun, could you please kindly take a look at the PR to respond to [the comment](https://github.com/lbl-srg/modelica-buildings/pull/3554#pullrequestreview-1700727099).

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Effectiveness.mo:
```diff
@@ -1,108 +1,99 @@
+  parameter Modelica.Units.SI.Efficiency epsLatCoo_nominal(final max=1) = 0.8
     "Nominal latent heat exchanger effectiveness at the cooling mode";
-  parameter Modelica.Units.SI.Efficiency epsS_cool_partload(max=1) = 0.75
+  parameter Modelica.Units.SI.Efficiency epsSenCoo_ParLoa(final max=1) = 0.75
```
We may change the parameter name to `epsSenCooPL`, see for example the variable `etaPL` in `Buildings.Fluid.Chillers.BaseClasses.Carnot`. And also change in the comment the `Partial load` to `Part load`.

Please apply the pattern to other similar parameters.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Effectiveness.mo:
```diff
@@ -1,108 +1,99 @@
-       x2 = epsL,
-       deltaX = 1E-7);
+   Real epsSen_ParLoa
+   "The partial load (75%) sensible heat exchanger effectiveness used for calculation";
```
change to `epsSenPL` and in the comment change the `partial load` to `part load`.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/Effectiveness.mo:
```diff
@@ -68,22 +78,25 @@ Buildings.Fluid.HeatExchangers.AirToAirHeatRecovery.BaseClasses.Effectiveness</a
 during the period from <i>60s</i> to <i>120s</i>.
-</li>
+</li> 
 </ul>
```
How should the outputs look like? It may be more helpful to we can add some description of the outputs.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/HeatExchangerWithInputEffectiveness.mo:
```diff
@@ -0,0 +1,135 @@
+  "Test model for the heat exchanger with input effectiveness"
+  extends Modelica.Icons.Example;
+  package Medium1 = Buildings.Media.Air
+     "Medium of the supply air";
```
`Supply Air`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/HeatExchangerWithInputEffectiveness.mo:
```diff
@@ -0,0 +1,135 @@
+  package Medium1 = Buildings.Media.Air
+     "Medium of the supply air";
+  package Medium2 = Buildings.Media.Air
+     "Medium of the exhaust air";
```
`Exhaust air`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/HeatExchangerWithInputEffectiveness.mo:
```diff
@@ -0,0 +1,135 @@
+    p(displayUnit="Pa") = 101325,
+    T=273.15 + 10,
+    nPorts=1)
+    "Sink that represents the ambient environment"
```
`Ambient`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/HeatExchangerWithInputEffectiveness.mo:
```diff
@@ -0,0 +1,135 @@
+    p(displayUnit="Pa") = 101325 + 100,
+    T=566.3,
+    nPorts=1)
+    "Source of the exhaust air"
```
`Exhaust air source`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/HeatExchangerWithInputEffectiveness.mo:
```diff
@@ -0,0 +1,135 @@
+    height=10,
+    duration=60,
+    offset=273.15 + 30,
+    startTime=120) "Temperature of the supply air"
```
`Supply air temperature`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/HeatExchangerWithInputEffectiveness.mo:
```diff
@@ -0,0 +1,135 @@
+    X={0.012,1 - 0.012},
+    p(displayUnit="Pa") = 1E5 - 110,
+    nPorts=1)
+    "Sink of the supply air"
```
`Supply air sink`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Validation/HeatExchangerWithInputEffectiveness.mo:
```diff
@@ -0,0 +1,135 @@
+    use_T_in=true,
+    p(displayUnit="Pa") = 100000,
+    nPorts=1)
+    "Source of the supply air"
```
`Supply air source`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/Examples/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,135 @@
+within Buildings.Fluid.HeatExchangers.AirToAirHeatRecovery.Examples;
+model WheelWithBypassDamper
```
Update the instances comment as above.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/Examples/WheelWithVariableSpeed.mo:
```diff
@@ -0,0 +1,132 @@
+    epsLatHea_nominal=0.7,
+    epsLatHea_ParLoa=0.6) "Wheel"
+    annotation (Placement(transformation(extent={{6,-4},{26,16}})));
+    Modelica.Blocks.Sources.Ramp WheSpe(
```
change `WheSpe` to `wheSpe`.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Effectiveness.mo:
```diff
@@ -113,47 +104,47 @@ air stream and the exhaust air stream.
-  epsS = y * (epsS_75 + (epsS_100 - epsS_75) * (vRat - 0.75)/0.25),
-  epsL = y * (epsL_75 + (epsL_100 - epsL_75) * (vRat - 0.75)/0.25),
+  epsSen = wheSpe * (epsSen_ParLoa + (epsSen_nominal - epsSen_ParLoa) * (rat - 0.75)/0.25),
+  epsLat = wheSpe * (epsLat_ParLoa + (epsLat_nominal - epsLat_ParLoa) * (rat - 0.75)/0.25),
```
change the `epsSen_ParLoa` to be consistent with the one in script.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+    annotation (Placement(transformation(extent={{70,-10},{90,10}})));
+  Modelica.Fluid.Interfaces.FluidPort_a port_a1(
+    redeclare final package Medium = Medium1)
+    "Fluid connector a1 of the supply air"
```
update the comments to be like `Fluid connector a1 of the supply air (positive design flow direction is from port_a1 to port_b1)`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+        iconTransformation(extent={{-110,50},{-90,70}})));
+  Modelica.Fluid.Interfaces.FluidPort_b port_b2(
+    redeclare final package Medium = Medium2)
+    "Fluid connector b2 of the supply air"
```
Is the comment correct? I thought the `port_b2` is the outlet of the exhaust air.

If my understanding is correct, then the comment should be `Fluid connector b2 of the exhaust air (positive design flow direction is from port_a2 to port_b2)`.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+  Modelica.Blocks.Sources.Constant uni(final k=1)
+    "Unity signal"
+    annotation (Placement(transformation(extent={{-154,32},{-140,46}})));
+  Modelica.Blocks.Math.Add sub(final k1=-1, final k2=+1)
```
It might be easier to use the `Buildings.Controls.OBC.CDL.Reals.Subtract`. And change the comment to be as `Difference of the two inputs`.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+Documentation(info="<html>
+<p>
+Model of a generic, sensible, and latent air-to-air heat recovery wheel, that consists of 
+a heat exchanger and supply/exhaust airflow bypass dampers.
```
`Model of a generic, sensible and latent air-to-air heat recovery wheel. It consists of a heat exchanger and two dampers to bypass the supply and exhaust airflow. The operation of the heat recovery wheel is controlled through adjusting the bypassing supply and exhaust airflow.`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+</p>
+<p>
+The operation of the heat recovery wheel is adjustable via bypassing supply/exhaust air 
+through the heat exchanger.
```
I have moved the section in the first paragraph.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+  "Sensible and latent air-to-air heat recovery wheel with bypass dampers"
+  replaceable package Medium1 =
+    Modelica.Media.Interfaces.PartialCondensingGases
+    "Medium of the supply air stream";
```
`Supply air`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+    annotation (Placement(transformation(extent={{-100,0},{-80,20}})));
+  Buildings.Fluid.Sensors.VolumeFlowRate VExh_flow(
+   redeclare package Medium = Medium2, final m_flow_nominal=m2_flow_nominal)
+    "Flow sensor in the exhaust air stream"
```
`Exhaust airflow sensor`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithBypassDamper.mo:
```diff
@@ -0,0 +1,296 @@
+        p=port_a1.p,
+        h=inStream(port_a1.h_outflow),
+        X=inStream(port_a1.Xi_outflow))))
+    "Temperature of the supply air"
```
`Supply air temperature`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithVariableSpeed.mo:
```diff
@@ -0,0 +1,227 @@
+    Modelica.Media.Interfaces.PartialCondensingGases
+    "Medium of the exhaust air stream";
+  parameter Modelica.Units.SI.MassFlowRate m1_flow_nominal
+    "Air flow rate of the supply air stream";
```
`Nominal supply air mass flow rate`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithVariableSpeed.mo:
```diff
@@ -0,0 +1,227 @@
+  parameter Modelica.Units.SI.MassFlowRate m1_flow_nominal
+    "Air flow rate of the supply air stream";
+  parameter Modelica.Units.SI.MassFlowRate m2_flow_nominal
+    "Air flow rate of the exhaust air stream";
```
`Nominal exhaust air mass flow rate`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithVariableSpeed.mo:
```diff
@@ -0,0 +1,227 @@
+    annotation (Placement(transformation(extent={{70,-10},{90,10}})));
+  Modelica.Fluid.Interfaces.FluidPort_a port_a1(
+    redeclare final package Medium = Medium1)
+    "Fluid connector a1 of the supply air"
```
Please update the comments of the fluid ports.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithVariableSpeed.mo:
```diff
@@ -0,0 +1,227 @@
+    final m_flow_nominal=m1_flow_nominal)
+    "Flow sensor in the supply air stream"
+    annotation (Placement(transformation(extent={{-42,92},{-28,108}})));
+  Modelica.Blocks.Sources.RealExpression TSup(final y=Medium1.temperature(
```
Can we use the temperature sensor, rather than using the real expression which is not visually clear as the connection? You may also change the exhaust air temperature measurement.

Please also change it in other model.

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithVariableSpeed.mo:
```diff
@@ -0,0 +1,227 @@
+Model of a generic, sensible, and latent air-to-air heat recovery wheel with a variable speed drive.
+</p>
+<p>
+The input requires no geometric data. Performance is defined by specifying sensible and latent effectiveness 
```
Do you mean `The model does not require geometric data.`?

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/WheelWithVariableSpeed.mo:
```diff
@@ -0,0 +1,227 @@
+        coordinateSystem(preserveAspectRatio=false, extent={{-100,-100},{100,140}})),
+Documentation(info="<html>
+<p>
+Model of a generic, sensible, and latent air-to-air heat recovery wheel with a variable speed drive.
```
`Model of a generic, sensible and latent air-to-air heat recovery wheel, which has the wheel speed as the input to control the heat recovery.`

JayHuLBL commented on Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/package.mo:
```diff
@@ -1,10 +1,9 @@
+annotation (Documentation(info="<html>
 <p>
-This package contains components models for air-to-air energy recovery devices.
+This package contains component models for air-to-air energy recovery devices.
```
change the `energy recovery devices` to `heat recovery devices`. It will be consistent throughout the entire package.

JayHuLBL reviewed (changes requested):
@SenHuang19 Please see the inline comments. Also please add the release note in `Buildings.UsersGuide.ReleaseNotes.Version_11_0_0`.

SenHuang19 commented:
> @SenHuang19 Please see the inline comments. Also please add the release note in `Buildings.UsersGuide.ReleaseNotes.Version_11_0_0`.

@JayHuLBL  thanks so much for your comments.
I addressed all your comments except
> Can we use the temperature sensor, rather than using the real expression which is not visually clear as the connection? You may also change the exhaust air temperature measurement.
Please also change it in other model.

I used the real expression based on `Buildings.Fluid.HeatExchangers.CoolingTowers.Merkel`. Please let me know if this is desired.

I have added the followings to `Buildings.UsersGuide.ReleaseNotes.Version_11_0_0`.

![image](https://github.com/lbl-srg/modelica-buildings/assets/6975889/d72fb638-a65e-4f0a-8716-3e76aefb4849)
